openssh/files/sshd_config

```diff
@@ -17,14 +17,33 @@ ListenAddress {{ server.bind.address }}
 
 Protocol 2
 # HostKeys for protocol version 2
+HostKey /etc/ssh/ssh_host_ed25519_key
 HostKey /etc/ssh/ssh_host_rsa_key
+{%- if server.get('dsa_enabled', false) %}
 HostKey /etc/ssh/ssh_host_dsa_key
+{%- endif %}
 {%- if grains.os_family != 'CentOS' %}
 HostKey /etc/ssh/ssh_host_ecdsa_key
 {%- endif %}
 #Privilege Separation is turned on for security
 UsePrivilegeSeparation yes
 
+{%- if server.get('kex_algorithms', False) %}
+KexAlgorithms {{ server.kex_algorithms|join(',') }}
+{%- else %}
+KexAlgorithms curve25519-sha256@libssh.org,ecdh-sha2-nistp521,ecdh-sha2-nistp384,ecdh-sha2-nistp256,diffie-hellman-group-exchange-sha256
```

Comment:
is that backward compatible? If/else would be enough (define best practice metadata under /metadata/service)